core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java

```diff
@@ -101,9 +104,11 @@ public void filter(ContainerRequestContext request)
                 return;
             }
             String principalName = (String) principal;
-            setAuthenticatedIdentity(request, Identity.forUser(userMapping.mapUser(principalName))
-                    .withPrincipal(new BasicPrincipal(principalName))
-                    .build());
+            Identity.Builder identity = Identity.forUser(userMapping.mapUser(principalName));
+            identity.withPrincipal(new BasicPrincipal(principalName));
+            groupsField.flatMap(field -> Optional.ofNullable((List<String>) claims.get().get(field)))
+                    .ifPresent(groups -> identity.withGroups(ImmutableSet.copyOf(groups)));
+            setAuthenticatedIdentity(request, identity.build());
```

Comment:
Maybe not for this PR, but would it be possible to make this filter using `OAuth2Authenticator`? Seems they share more logic now than previously.

Reply:
I initially thought about it and I think it's generally a good idea but it would require a bit broader refactor. The important difference between `OAuth2Authenticator` and this filter is that protocol authenticator throws an Authentication exception (`401 Unauthorized`) whereas this filter redirects the browser to authentication url (`303 See Other`). It's doable ofc but as I said, it requires a bit of refactoring.